runtime/js/40_spawn.js

```diff
@@ -119,8 +125,12 @@
         this.#stderr = readableStreamForRid(stderrRid);
       }
 
+      const onAbort = () => this.kill("SIGTERM");
+      signal?.[add](onAbort);
+
       this.#status = core.opAsync("op_spawn_wait", this.#rid).then((res) => {
         this.#rid = null;
+        signal?.[remove](onAbort);
```

Comment:
Should this be done in a `.finally(...)` so this gets unsubscribed if it ever throws?

Reply:
maybe, though I dont think there is a scenario this would throw, and if it does, then i'd consider that a bug to some degree